components/chainhook-sdk/src/observer/http.rs

```diff
@@ -262,99 +234,102 @@ pub fn handle_new_microblocks(
+                        ctx,
+                    );
                 }
             };
         }
         Ok(None) => {
-            ctx.try_log(|logger| slog::warn!(logger, "unable to infer chain progress"));
+            try_info!(ctx, "No chain event was generated");
         }
         Err(e) => {
-            ctx.try_log(|logger| slog::error!(logger, "unable to handle stacks microblock: {}", e));
+            return error_response(format!("Chain event error: {e}"), ctx);
         }
     }
 
-    Json(json!({
-        "status": 200,
-        "result": "Ok",
-    }))
+    success_response()
 }
 
 #[post("/new_mempool_tx", format = "application/json", data = "<raw_txs>")]
 pub fn handle_new_mempool_tx(
     raw_txs: Json<Vec<String>>,
     background_job_tx: &State<Arc<Mutex<Sender<ObserverCommand>>>>,
     ctx: &State<Context>,
-) -> Json<JsonValue> {
-    ctx.try_log(|logger| slog::debug!(logger, "POST /new_mempool_tx"));
-    let transactions = raw_txs
+) -> Result<Json<JsonValue>, Custom<Json<JsonValue>>> {
+    try_info!(ctx, "POST /new_mempool_tx");
+    let transactions = match raw_txs
         .iter()
         .map(|tx_data| {
-            let (tx_description, ..) = indexer::stacks::get_tx_description(tx_data, &vec![])
-                .expect("unable to parse transaction");
-            MempoolAdmissionData {
-                tx_data: tx_data.clone(),
-                tx_description,
-            }
+            indexer::stacks::get_tx_description(tx_data, &vec![])
+                .map(|(tx_description, ..)| MempoolAdmissionData {
+                    tx_data: tx_data.clone(),
+                    tx_description,
+                })
+                .map_err(|e| e)
         })
-        .collect::<Vec<_>>();
+        .collect::<Result<Vec<MempoolAdmissionData>, _>>()
+    {
+        Ok(transactions) => transactions,
+        Err(e) => {
+            return error_response(format!("Failed to parse mempool transactions: {e}"), ctx);
+        }
+    };
 
     let background_job_tx = background_job_tx.inner();
-    if let Ok(tx) = background_job_tx.lock() {
-        let _ = tx.send(ObserverCommand::PropagateStacksMempoolEvent(
-            StacksChainMempoolEvent::TransactionsAdmitted(transactions),
-        ));
-    };
+    match background_job_tx.lock() {
+        Ok(tx) => {
+            match tx.send(ObserverCommand::PropagateStacksMempoolEvent(
+                StacksChainMempoolEvent::TransactionsAdmitted(transactions),
+            )) {
+                Ok(_) => {}
+                Err(e) => {
+                    return error_response(
+                        format!("Unable to send stacks chain event: {}", e.to_string()),
+                        ctx,
+                    );
+                }
+            }
+        }
+        Err(e) => {
+            return error_response(
+                format!("Unable to acquire background_job_tx: {}", e.to_string()),
+                ctx,
+            );
+        }
+    }
```

Comment:
Please check the comment above about ignoring success cases and nested statements.